x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/utils/indexing_service.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getHttp } from '../../../../kibana_services';
+import { INDEX_SOURCE_API_PATH } from '../../../../../common';
+
+export const createNewIndexAndPattern = async (indexName: string) => {
+  return await getHttp().fetch({
```

Comment:
Can you use generics to type the response, like `await getHttp().fetch<CreateDocSourceResp>`